flask_bootstrap/templates/bootstrap/form.html

```diff
@@ -1,231 +1,227 @@
-    {%- if enctype is none -%}
-        {%- for field in form %}
-            {%- if field.type == 'FileField' %}
-                {#- for loops come with a fairly watertight scope, so this list-hack is
-          used to be able to set values outside of it #}
-                {%- set _ = _enctype.append('multipart/form-data') -%}
-            {%- endif %}
-        {%- endfor %}
-    {%- else %}
-        {% set _ = _enctype.append(enctype) %}
-    {%- endif %}
-    <form{%- if action != None %} action="{{ action }}"{% endif -%}
-            {%- if id %} id="{{ id }}"{% endif -%}
-            {%- if method %} method="{{ method }}"{% endif %}
-            class="form
-{%- if extra_classes %} {{ extra_classes }}{% endif -%}{%- if form_type == "inline" %} form-inline{% endif -%}"
-            {%- if _enctype[0] %} enctype="{{ _enctype[0] }}"{% endif -%}
-            {%- if role %} role="{{ role }}"{% endif -%}
-            {%- if novalidate %} novalidate{% endif -%}
-            {%- if render_kw %} {{ render_kw|xmlattr }}{% endif -%}>
-        {{ form.hidden_tag() }}
-        {{ form_errors(form, hiddens='only') }}
-        {%- for field in form %}
-            {% if not bootstrap_is_hidden_field(field) -%}
-                {{ render_field(field,
-                    form_type=form_type,
-                    horizontal_columns=horizontal_columns,
-                    button_map=button_map) }}
-            {%- endif %}
-        {%- endfor %}
-    </form>
-{%- endmacro %}
\ No newline at end of file
+    its BSD License. Copyright (c) 2013, Marc Brinkmann. All rights reserved. #}
+{%- from 'bootstrap/utils.html' import typename, modulename, class, get_col_from_count, get_col, tag -%}
+
+
+{%- macro css_horizontal_fix() -%}
+{%- block css_horizontal_fix scoped -%}
+    <style>
+        /* A fix for custom checkboxes and custom files in horizontal rows */
+        form .form-row > [class*="col-"].custom-control .custom-control-label::before,
+        form .form-row > [class*="col-"].custom-control .custom-control-label::after,
+        form .form-row > [class*="col-"].custom-file .custom-file-label {
+            margin-left: 5px;
+            margin-right: 5px;
+        }
+        form .row > [class*="col-"].custom-control .custom-control-label::before,
+        form .row > [class*="col-"].custom-control .custom-control-label::after,
+        form .row > [class*="col-"].custom-file .custom-file-label {
+            margin-left: 15px;
+            margin-right: 15px;
+        }
+        form .custom-control {
+            padding-left: 0px;
+        }
+        form .row > [class*="col-"].custom-control {
+            padding-left: 15px;
+        }
+        form .custom-control .custom-control-label {
+            margin-left: 1.5rem;
+        }
+    </style>
+{%- endblock -%}
+{%- endmacro -%}
+
+
+{%- macro hidden_errors(field) -%}
+{%- block hidden_errors scoped -%}
+    {# Doesn't actually work, has no element which makes it "display: block" #}
```

Comment:
Also regarding this, do you know how the preffered way of displaying errors like this is in bootstrap? We can't use the normal `<span class="invalid-feedback">`, since that's specific to form elements. Something like `<div class="flash alert alert-danger">` perhaps?

Reply:
Use Alert will be good. Why adding a `flash` class?

Reply:
I'm not really sure why I thought that, I probably just looked at some legacy code ;)